graylog2-shared/src/main/java/org/graylog2/shared/buffers/processors/DecodingProcessor.java

```diff
@@ -76,17 +81,20 @@ public void onEvent(MessageEvent event, long sequence, boolean endOfBatch) throw
         try {
             // always set the result of processMessage, even if it is null, to avoid later stages to process old messages.
             // basically this will make sure old messages are cleared out early.
-            event.setMessage(processMessage(event.getRaw()));
+            event.setMessages(processMessage(event.getRaw()));
```

Comment:
Changing it this way will force two new object allocations for each message (and single messages are the norm).
Could we refactor this so that we don't have to do those two allocations and simply change the `ProcessBufferProcessor#onEvent` to figure out whether to invoke the `MessageFilter` more than once?